authors/apps/articles/views.py

```diff
@@ -0,0 +1,57 @@
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+from .models import Article
+from django.shortcuts import get_object_or_404
+from rest_framework import serializers
+from .serializers import ArticleSerializer, ArticleUpdateSerializer
+from .renderers import ArticleJSONRenderer
+from rest_framework import status
+from rest_framework.response import Response
+from authors.apps.utils.messages import error_messages
+from authors.apps.profiles.models import Profile
+
+
+class ArticleListCreateView(generics.ListCreateAPIView):
+    queryset = Article.objects.all()
+    serializer_class = ArticleSerializer
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleJSONRenderer,)
+
+    def perform_create(self, serializer):
+        serializer.save(
+            author=Profile.objects.get(user=self.request.user)
+        )
+
+
+class ArticleUpdateDeleteView(generics.RetrieveUpdateDestroyAPIView):
+    """ Updates and deletes an article instance """
+    serializer_class = ArticleUpdateSerializer
+    permission_class = (IsAuthenticated,)
+
+    def get_object(self):
+        slug = self.kwargs.get('slug')
+        return get_object_or_404(Article, slug=slug)
+
+    def check_if_is_author(self):
+        self.article = self.get_object()
+        if str(self.article.author) != str(self.request.user.username):
+            return True
+        return False
+
+    def perform_update(self, serializer):
+        if self.check_if_is_author():
+            raise serializers.ValidationError(
+                 error_messages['permission_update_msg']
+                )
+        serializer.save(
+         author=Profile.objects.get(user=self.request.user)
+        )
+
+    def destroy(self, *args, **kwargs):
+        instance = self.get_object()
+        if self.check_if_is_author():
+            raise serializers.ValidationError(
+                    error_messages['permission_delete_msg'])
+        self.perform_destroy(instance)
+        return Response({"message": "Article successfully deleted"},
```

Comment:
@mariamiah put the messages in the messages.py file and use place holders e.g.
```suggestion
        "delete_message": "{} successfully deleted"
```

so that it can be re-used.